File: MediCareWebApi/Models/Appointment.cs
using MediCare.Enums;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MediCare.Models
{
	public class Appointment : Entity
	{
		[Required]
		public DateTime Time { get; set; }

		[Required]
		public AppointmentStatus Status { get; set; }

		[MaxLength(256)]
		public string? Diagnosis { get; set; }

		[Required]
		public int PatientsUserId { get; set; }

		[ForeignKey("PatientsUserId")]
		public Patient Patient { get; set; }

		[Required]
		public int DoctorsUserId { get; set; }

		[ForeignKey("DoctorsUserId")]
		public Doctor Doctor { get; set; }

		[Required]
		public int ServiceId { get; set; }

		[ForeignKey("ServiceId")]
		public Service Service { get; set; }

		public ICollection<Feedback> Feedbacks { get; set; } = new List<Feedback>();
	}
}

File: MediCareWebSpa/src/app/DTOs/models/appointment.ts
import { AppointmentStatus } from "../../enums/appointment-status";
import { Doctor } from "./doctor";
import { Patient } from "./patient";
import { Service } from "./service";

export class Appointment {
  id: number;
  time: Date;
  status: AppointmentStatus;
  diagnosis: string;
  patientsUserId: number;
  patient: Patient;
  doctorsUserId: number;
  doctor: Doctor;
  serviceId: number;
  service: Service;

  constructor(
    id: number,
    time: Date,
    status: AppointmentStatus,
    diagnosis: string,
    patientsUserId: number,
    patient: Patient,
    doctorsUserId: number,
    doctor: Doctor,
    serviceId: number,
    service: Service
  ) {
    this.id = id;
    this.time = time;
    this.status = status;
    this.diagnosis = diagnosis;
    this.patientsUserId = patientsUserId;
    this.patient = patient;
    this.doctorsUserId = doctorsUserId;
    this.doctor = doctor;
    this.serviceId = serviceId;
    this.service = service;
  }
}
